Cython/Compiler/Pipeline.py

```diff
@@ -21,6 +22,46 @@ def abort_on_errors(node):
         raise AbortError("pipeline break")
     return node
 
+def create_shared_library_pipeline(context, scope, options, result):
+
+    def generate_tree_factory(context):
+        def generate_tree(compsrc):
+            from . import MemoryView
+            import Cython
+
+            # Force to generate __Pyx_ExportFunction
+            Options.cimport_from_pyx = True
+
+            source_desc = compsrc.source_desc
+            full_module_name = compsrc.full_module_name
+
+            initial_pos = (source_desc, 1, 0)
+            scope = context.find_module(full_module_name, pos = initial_pos, need_pxd = 0)
+
+            tree = context.parse(source_desc, scope, pxd = 0, full_module_name = full_module_name)
+
+            tree.is_pxd = False
+            tree.compilation_source = compsrc
+            tree.scope = scope
+
+            scope.use_utility_code(MemoryView.memoryview_utility_code)
+
+            scope.use_utility_code(MemoryView.memviewslice_init_code)
+            scope.use_utility_code(MemoryView.typeinfo_to_format_code)
```

Comment:
This seems a good time to process the Cython utility code, rather than using a global constant to prepare it.